### PR TITLE
4.0.0 dev compatibile

### PR DESCRIPTION
**Pull request by KarishmaBothara**
With this PR, 

1. Api is updated with current latest polkadot-version 7.6.1, 
2. Updated types and test for nft as per (https://github.com/cennznet/cennznet/pull/535/files)
3. Now polkadots 7.x typegen script requires endpoint
generate:defs": "ts-node --skip-project node_modules/.bin/polkadot-types-from-defs --package @cennznet/types/interfaces --endpoint ./cennznet.json   --input ./src/interfaces"  (https://github.com/polkadot-js/api/issues/4481)
4. Updated staking derive module to stakingCennznet cause as per new polkadot version  https://github.com/polkadot-js/api/blob/master/packages/api-derive/src/bundle.ts#L67
https://github.com/polkadot-js/api/blob/master/packages/api-derive/src/bundle.ts#L23 (enable derive only if erasRewardPoints exist
5. Updated WithdrawReasons to Reasons for BalanceLock

### CODE REVIEW

**Comment by KarishmaBothara** [comment]
Things to do 
1. Unskip tests related to custom/slim metadata (needs some investigation - (FATAL: Unable to initialize the API: createType(Lookup111):: PortableRegistry: Unable to find type with lookupId Lookup111)
2. Bring in changes related to Eth-bridge from master to develop branch and test

**Review by ken-centrality — APPROVED** [review]
@KarishmaBothara can't say that I fully understand this PR but I reckon it's good to merge and release the next alpha version to try it out.

**Comment by KarishmaBothara** [comment]
Tests related to custom/slim metadata are all working
Eth-bridge stuff is added to this branch

**Review by holygits — APPROVED** [review]
lgtm in general.
few commented code lines to tidy up
can you take note (for release notes/upgrade guide) of the breaking changes e.g. `api.derive.staking => api.derive.cennzStaking`, are users required to update to node 16 etc.

**Comment by KarishmaBothara** [comment]
> lgtm in general.
> few commented code lines to tidy up
> can you take note (for release notes/upgrade guide) of the breaking changes e.g. `api.derive.staking => api.derive.cennzStaking`, are users required to update to node 16 etc.

I am keeping a note of all the breaking changes and projects that would need to be updated for example (cennznet.io)
Since we are using the latest polkadot version it requires us to use one of the following
The engine "node" is incompatible with this module. Expected version "^12.22.0 || ^14.17.0 || >=16.0.0". Got "12.21.0"

File: packages/api/test/e2e/api.create.e2e.ts [hunk 1]
```diff
@@ -109,7 +113,7 @@ describe('e2e api create', () => {
   });
 
   it('Should connect to all available networks on cennznet via network name', async done => {
-    const networkNames = ['azalea', 'nikau', 'rata', 'local'] as const;
+    const networkNames = ['local'] as const;
```
[the remark this answers is not in the document]
Comment (KarishmaBothara) [reply packages/api/test/e2e/api.create.e2e.ts]: changed it back

**Pull request merged**